narwhals/functions.py

```diff
@@ -42,6 +48,61 @@ def concat(
     )
 
 
+def from_dict(data: dict[str, Any], *, native_namespace: Any) -> DataFrame[Any]:
```

Comment:
Shall we add just the schema argument? We should be able to convert it to the native via `reverse_translate_dtype`

Reply:
yes! not as required though, right?

Reply:
yep, polars default is None, it will infer

Reply:
If so, OK to leave it as a follow-up? This is all I need for Formulaic at the moment 😄

https://github.com/narwhals-dev/narwhals/issues/603